Remove additional endpoint statuses in CRD (#30761)
cilium-team/cilium: Documentation/operations/upgrade.rst

~~~~~~~~~~~~~~~

* The unused flag ``sidecar-istio-proxy-image`` has been removed.

> * The flag ``endpoint-status`` has been removed.

Reviewer: I think we still have some docs referring to the endpoint status. We should remove or update those sections:
https://docs.cilium.io/en/stable/security/aws/#validate-that-derived-policy-is-in-place (last paragraph)

Author: Hmm good point. I guess it didn't work before without setting `endpoint-status=policy` explicitly, which the doc doesn't mention :confused: 

Anyway, I will replace it with `cilium-dbg` command.

Author: ok, I've updated this doc. The new command doesn't look great, but at least it works as compared to the previous one.

>   Please take a look at helm option deprecation for more information.

Reviewer 2: Slight rewording suggestion:

```suggestion
  More information can be found in the following Helm upgrade notes.
```

Author: Done.

* Deprecated options ``proxy.prometheus.enabled`` and ``proxy.prometheus.port`` have been removed.
  Please use ``envoy.prometheus.enabled`` and ``envoy.prometheus.port`` instead.
* The unused helm option ``proxy.sidecarImageRegex`` has been removed.

> * Helm value ``endpointStatus`` has been removed. Instead of relying on additional statuses in CiliumEndpoints CRD,
>   please rely on Cilium's metrics to monitor status of endpoints, like ``cilium_policy``, ``cilium_policy_endpoint_enforcement_status``, ``cilium_controllers_failing``, ``cilium_endpoint_state``.
>   More detailed information about specific endpoint status is still available through ``cilium-dbg endpoint get``.

Reviewer 2: Grammar and word choice suggestions:

```suggestion
* The helm option ``endpointStatus`` has been removed. Instead of relying on additional statuses in the CiliumEndpoints CRD,
  rely on Cilium's metrics to monitor the status of endpoints. Example metrics include: ``cilium_policy``, ``cilium_policy_endpoint_enforcement_status``, ``cilium_controllers_failing`` and ``cilium_endpoint_state``.
  More detailed endpoint status information is still available through ``cilium-dbg endpoint get``.
```

Author: Done.
